### PR TITLE
MGDOBR-345: add openapi spec and add check in CI pipeline

### PR DESCRIPTION
**Pull request by r00ta**
[MGDOBR-345](https://issues.redhat.com/browse/MGDOBR-345) 

Please make sure that your PR meets the following requirements:

- [x] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config)
- [x] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [x] All new functionality is tested
- [x] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

* <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

* <b>Rebase the pull request</b>  
  Comment with: `/rebase`.

</details>

### CODE REVIEW

**Comment by manstis** [comment]
@robpblake Could you please add Roberto Emanuel to this repo (so I can add him to this PR)?

We will be generating the API to use for the UI from the OpenAPI definition too.

**Comment by r00ta** [comment]
> @robpblake Could you please add Roberto Emanuel to this repo (so I can add him to this PR)?
> 
> We will be generating the API to use for the UI from the OpenAPI definition too.

Done, the invite should be with Roberto

**Comment by manstis** [comment]
@kelvah Yay, you're here.. anyway, FYI.

File: .github/workflows/CI.yaml [hunk 1]
```diff
@@ -29,3 +29,8 @@ jobs:
           check_name: Test Results
           fail_on_test_failures: true
           fail_if_no_tests: false
+      - name: Validate openapi spec
+        if: ${{ always() }}
+        run: |
+          differences=$(diff openapi/openapi.yaml manager/target/generated/openapi.yaml)
+          if [ $? -ne 0 ]; then echo "Please move the manager/target/generated/openapi.yaml into openapi/openapi.yaml"; false ; fi
```
Comment (danielezonca) [inline .github/workflows/CI.yaml:36]: Just to double check, is this step enough to make the build fail?

**Review by danielezonca — APPROVED** [review]

File: .github/workflows/CI.yaml (hunk 1 above)
Comment (r00ta) [reply .github/workflows/CI.yaml]: Yep, `diff` will return status code `1` if the two files are different. Then the step will fail and the build would be red

**Pull request merged**